dashboard/test/lib/services/partial_registration/user_builder_test.rb

```diff
@@ -0,0 +1,280 @@
+      @user = Services::PartialRegistration::UserBuilder.call(request: @request)
+    end
+
+    assert_equal parent_email, @user.parent_email
+    assert_equal TEST_IP, @user.parent_email_preference_request_ip
+    assert_equal EmailPreference::ACCOUNT_SIGN_UP, @user.parent_email_preference_source
+    assert_equal '0', @user.parent_email_update_only
+    assert_equal 'yes', @user.parent_email_preference_opt_in
+    assert EmailPreference.where(email: @user.parent_email).last.opt_in
+  end
+
+  # Teacher tests
+  test 'builds teacher user with default values' do
+    setup_partial_user({user_type: 'teacher'})
+
+    assert_creates(User) do
+      @user = Services::PartialRegistration::UserBuilder.call(request: @request)
+    end
+
+    assert_equal TEST_USER_NAME, @user.name
+    assert_equal TEST_USER_EMAIL, @user.email
+    assert_equal '21+', @user.age.to_s
+    assert @user.terms_of_service_version
+
+    assert_equal TEST_IP, @user.email_preference_request_ip
+    assert_equal EmailPreference::ACCOUNT_SIGN_UP, @user.email_preference_source
+    assert_equal '0', @user.email_preference_form_kind
+    assert_equal 'no', @user.email_preference_opt_in
+    refute EmailPreference.where(email: @user.email).last.opt_in
+  end
+
+  test 'builds teacher user opted into marketing emails' do
+    setup_partial_user({user_type: 'teacher', email_preference_opt_in: true})
+
+    assert_creates(User) do
+      @user = Services::PartialRegistration::UserBuilder.call(request: @request)
+    end
+
+    assert_equal TEST_IP, @user.email_preference_request_ip
+    assert_equal EmailPreference::ACCOUNT_SIGN_UP, @user.email_preference_source
+    assert_equal '0', @user.email_preference_form_kind
+    assert_equal 'yes', @user.email_preference_opt_in
+    assert EmailPreference.where(email: @user.email).last.opt_in
+  end
+
+  test 'builds teacher with nces school in our database' do
+    fake_school = create :school
+    fake_school_info = create :school_info, school_id: fake_school.id
+    setup_partial_user({user_type: 'teacher', school_info_attributes:
+      {
+        schoolId: fake_school.id,
+        schoolName: fake_school.name,
+        schoolType: fake_school.school_type,
+        schoolZip: fake_school.zip,
+        schoolState: fake_school.state,
+        country: fake_school_info.country,
+        fullAddress: fake_school_info.full_address
+      }}
+    )
+
+    assert_creates(User) do
+      @user = Services::PartialRegistration::UserBuilder.call(request: @request)
+    end
+
+    assert_equal fake_school_info.id, @user.school_info_id
+    assert UserSchoolInfo.where(school_info_id: fake_school_info.id, user_id: @user.id)
+  end
+
+  test 'builds teacher that adds their own US school' do
+    school_name = 'Fake US School Name'
+    school_zip = '11111'
+    school_country = 'US'
+    setup_partial_user({user_type: 'teacher', school_info_attributes:
+      {
+        schoolId: SharedConstants::NON_SCHOOL_OPTIONS.CLICK_TO_ADD,
```

Comment:
Is the new sign up flow sending a `CLICK_TO_ADD` id? Or is this just to assert that the `school_id` ends up as nil on line 210? I only ask because other areas where school info inputs are used I've seen logic that removes the `school_id` property if it's not an actual nces school id